src/nlsolve/nlsolve.jl

```diff
@@ -42,6 +42,10 @@ function nlsolve!(nlsolver::AbstractNLSolver, integrator)
     apply_step!(nlsolver, integrator)
 
     # check for convergence
+    if iter == 1 && ndz < 1e-3
```

Comment:
What's the reason for using the cutoff `1e-3` here? It seems a bit arbitrary and unfortunately leads to a bunch of test errors.

Couldn't this be covered by the existing check below? E.g., by changing it to
```
if η * ndz < κ && !(isnewton(nlsolver) && iszero(integrator.success_iter))
```
? It seems the check `iszero(ndz)` is not needed anyway?

Reply:
The cutoff is somewhat arbitrary, but it is already scaled by tolerance so it's just a relative parameter. The difficulty here is that if it doesn't exist, then really good guesses cause a convergence failure because then the rate is essentially zero